### PR TITLE
feat: add typstyle

### PR DESCRIPTION
**Pull request by leana8959**
Hello,

I have added `typstyle`, a formatter for `typst`.

Though, there's a little issue: it's not in the version of nixpkgs pinned by this project, and it won't work. What would be a good solution ?

Thanks and have a nice day :)

### CODE REVIEW

**Comment by leana8959** [comment]
_Psst, The CI failed because `typstyle` is not in the `nixpkgs` pinned in the flake._

File: nix/tools.nix [hunk 1]
```diff
@@ -71,6 +71,7 @@
 , treefmt
 , typos
 , typstfmt
+, typstyle
```
Comment (sandydoo) [inline nix/tools.nix:74]:
```suggestion
, typstyle ? null
```

File: modules/hooks.nix [hunk 2]
```diff
@@ -3414,6 +3414,13 @@ lib.escapeShellArgs (lib.concatMap (ext: [ "--ghc-opt" "-X${ext}" ]) hooks.ormol
         entry = "${hooks.typstfmt.package}/bin/typstfmt";
         files = "\\.typ$";
       };
+      typstyle = {
+        name = "typstyle";
+        description = "Beautiful and reliable typst code formatter";
+        packages = tools.typstyle;
+        entry = "${hooks.typstyle.package}/bin/typstyle";
```
Comment (sandydoo) [inline modules/hooks.nix:3421]: You can check whether `tools.typstyle` is `null` and print a nicer error message if it is. Search for `lib.throwIf` for examples.

**Review by sandydoo — CHANGES_REQUESTED** [review]
I'll bump nixpkgs so that this works out of the box.

**Comment by leana8959** [comment]
The requested changes were added :)

**Comment by sandydoo** [comment]
Thanks, @leana8959!

**Pull request merged**